commit 5f64b253db2b62313940108251deeacbc164472e
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:20 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-vcs-from-macports.diff

diff --git a/src/port1.0/portfetch.tcl b/src/port1.0/portfetch.tcl
index e3452ee53..9503032ec 100644
--- a/src/port1.0/portfetch.tcl
+++ b/src/port1.0/portfetch.tcl
@@ -65,7 +65,10 @@ commands svn
 default extract.suffix .tar.gz
 default fetch.type standard
 
-default bzr.cmd {[findBinary bzr $portutil::autoconf::bzr_path]}
+#default bzr.cmd {[findBinary bzr $portutil::autoconf::bzr_path]}
+# accessing ${prefix_frozen} directly here used to work, now we need a wrapper
+# (At least for building MacPorts?)
+default bzr.cmd {[portfetch::find_bzr_path]}
 default bzr.dir {${workpath}}
 default bzr.revision -1
 default bzr.pre_args {--builtin --no-aliases checkout --lightweight --verbose}
@@ -206,11 +209,16 @@ proc portfetch::set_fetch_type {option action args} {
 proc portfetch::find_git_path {args} {
     global prefix_frozen os.platform os.major
     # Oldest macOS version whose git can validate GitHub's SSL certificate.
-    if {${os.major} >= 14 || ${os.platform} ne "darwin"} {
-        return [findBinary git $portutil::autoconf::git_path]
-    } else {
+#     if {${os.major} >= 14 || ${os.platform} ne "darwin"} {
+#         return [findBinary git $portutil::autoconf::git_path]
+#     } else {
         return ${prefix_frozen}/bin/git
-    }
+#     }
+}
+
+proc portfetch::find_bzr_path {args} {
+    global prefix_frozen
+    return ${prefix_frozen}/bin/bzr
 }
 
 set_ui_prefix
